### ISSUE TITLE
Ordering of ensure / ensureOr arguments in MonadError

### ISSUE THREAD

**Issue by Mullefa** [typelevel/cats#2875]
Apologies if this has already been raised; I couldn't find any related Issues / PRs.

I'm just wondering what the reasoning is for the ordering of the arguments of `MonadError#ensure` and `MonadError#ensureOr`.

I think typical usage of them leads to code that reads confusingly e.g.
```scala
case object CustomerUnderage extends Exception
def isCustomerOfLegalAge(customer: Customer): Boolean = ???

// F has MonadError type class
val customer: F[Customer] = ???

// current API - reads confusingly i.e. ensure customer is underage ?!
customer.ensure(CustomerUnderage)(isCustomerOfLegalAge)
```

Instead, what are peoples' thoughts on something like:
```scala
// proposed API
customer.ensure(isCustomerOfLegalAge)(onError = CustomerUnderage)
```

**Comment by LukaJCB** [comment]
Hi @Mullefa thanks for bringing up this issue :)

I agree with the premise of your argument, though I'm not sure these small syntactical improvements are worth breaking backwards compatibility over. We could think about doing this for 3.0, but only with a ScalaFix that would automatically fix the breakage. Willing to discuss more

**Comment by kailuowang** [comment]
I agree that sequence is less consistent with how the logic is presented in a English sentence. However it's a convention to place the function parameter of a method/function in last parameter group and the motivation of that convention, I believe, is that it makes it more readable when you pass a relatively long anonymous function as a parameter. 

i.e. it's more readable to have
```scala
customer.ensure(CustormerUnderAge) { c => 
    // validation logic here 
}
```
than
```scala
customer.ensure { c => 
    // validation logic here 
} (CustormerUnderAge)
```
I agree that which is more readable is subjective, but people are not used to see a dangling parameter after a block of code, so that's also conventional.
